deeptime/markov/_observables.py

```diff
@@ -0,0 +1,66 @@
+from typing import Union
+
+import numpy as np
+
+from ..base import Observable
+
+
+class MembershipsObservable(Observable):
+
+    @staticmethod
+    def _to_markov_model(model):
+        if hasattr(model, 'prior'):
+            model = model.prior
+        if hasattr(model, 'transition_model'):
+            model = model.transition_model
+        from deeptime.markov.msm import MarkovStateModel
+        assert isinstance(model, MarkovStateModel), f"This should be a Markov state model but was {type(model)}."
+        return model
+
+    def __init__(self, test_model, memberships,
+                 initial_distribution: Union[str, np.ndarray] = 'stationary_distribution'):
+        from deeptime.markov import PCCAModel
+        self.memberships = memberships if not isinstance(memberships, PCCAModel) else memberships.memberships
+        self.n_states, self.n_sets = self.memberships.shape
+
+        msm = MembershipsObservable._to_markov_model(test_model)
+        n_states = msm.n_states
```

Comment:
Doubly defined var, `self.n_states` and `n_states` (equivalence asserted below from `self.memberships.shape`)